### ISSUE TITLE
Loss of context with certain dependencies

### ISSUE THREAD

**Issue by watson** [elastic/apm-agent-nodejs#406]
A user experienced a problem where transactions of type `request` randomly gets either the result `HTTP ?xx` type or `success`. This indicates a problem similar to elastic/apm-agent-nodejs#367 which most likely means that a user-land callback queue makes us loose the context *before* `res.writeHead` is called.

For more info see Discuss thread: https://discuss.elastic.co/t/difference-in-data-captured-by-apm-agent-6-2-3/136604/2

Here's the users dependencies:

```json
"dependencies": {
    "agentkeepalive": "^3.3.0",
    "elastic-apm-node": "^1.4.0",
    "ansi-regex": "^3.0.0",
    "ansi-styles": "^3.2.0",
    "asn1": "^0.2.3",
    "assert-plus": "^1.0.0",
    "auth-challenge-response": "^0.5.1",
    "aws-sign2": "^0.7.0",
    "aws4": "^1.6.0",
    "base64-arraybuffer": "^0.1.5",
    "bcrypt-pbkdf": "^1.0.1",
    "bignumber.js": "^5.0.0",
    "body-parser": "^1.18.2",
    "boom": "^7.1.1",
    "buffer": "^4.9.1",
    "buffer-shims": "^1.0.0",
    "caseless": "^0.12.0",
    "chalk": "^2.3.0",
    "cookie-parser": "^1.4.3",
    "core-js": "^2.5.3",
    "cryptiles": "^4.1.1",
    "dashdash": "^1.14.1",
    "delayed-stream": "^1.0.0",
    "double-ended-queue": "^2.1.0-0",
    "ecc-jsbn": "^0.1.1",
    "exec-php": "^0.0.3",
    "express": "^4.16.2",
    "extend": "^3.0.1",
    "extsprintf": "^1.4.0",
    "forever-agent": "^0.6.1",
    "form-data": "^2.3.1",
    "fs-extra": "^5.0.0",
    "generate-function": "^2.0.0",
    "generate-object-property": "^1.2.0",
    "generic-pool": "^2.5.4",
    "ioredis": "^3.2.2",
    "js-nacl": "^1.2.2",
    "json-schema": "^0.2.3",
    "json-stringify-safe": "^5.0.1",
    "jsonpointer": "^4.0.1",
    "jsprim": "^2.0.0",
    "loadtest": "^2.3.0",
    "mssql": "^4.1.0",
    "mysql": "^2.15.0",
    "node-uuid": "^1.4.8",
    "promise": "^7.3.1",
    "qrcode-js": "0.0.2",
    "tedious": "^1.14.0",
    "winston": "^2.4.0",
    "winston-daily-rotate-file": "^1.7.2"
}
```

**Comment by alvarolobato** [comment]
We think that this may have been fixed indirectly by another commit.

**Comment by watson** [comment]
I think this is fixed by #637. Closing for now.
